source/NanostackRfPhyAT86RF215.cpp

```diff
@@ -580,31 +591,36 @@ static void rf_handle_ack(uint8_t seq_number, uint8_t pending)
 
 static void rf_handle_rx_done(void)
 {
+    receiver_enabled = false;
     TEST_RX_DONE
     rf_backup_timer_stop();
     if (rf_state == RF_CSMA_WHILE_RX) {
         rf_state = RF_CSMA_STARTED;
-        rf_backup_timer_start(MAX_PACKET_SENDING_TIME);
+        uint32_t backup_time = tx_time - rf_get_timestamp();
+        // Next TX start event must occur in less than 65ms
+        if (backup_time > 65000) {
+            backup_time = 0;
+        }
+        rf_backup_timer_start(0, backup_time + CCA_BACKUP_TIMEOUT);
     } else if (rf_state == RF_RX_STARTED) {
         rf_state = RF_IDLE;
     }
     if (rf_read_bbc_register(BBC_PC, rf_module) & FCSOK) {
-        uint16_t rx_data_length = rf_read_rx_frame_length(rf_module);
-        if (!rf_read_rx_buffer(rx_data_length, rf_module)) {
+        if (!rf_read_rx_buffer(cur_rx_packet_len, rf_module)) {
             uint8_t version = ((rx_buffer[1] & VERSION_FIELD_MASK) >> SHIFT_VERSION_FIELD);
             if (((rx_buffer[0] & MAC_FRAME_TYPE_MASK) == MAC_TYPE_ACK) && (version < MAC_FRAME_VERSION_2)) {
                 rf_handle_ack(rx_buffer[2], rx_buffer[0] & MAC_DATA_PENDING);
             } else {
                 int8_t rssi = (int8_t) rf_read_rf_register(RF_EDV, rf_module);
                 if (phy_current_config.modulation == M_OQPSK) {
-                    rx_data_length -= 2;
+                    cur_rx_packet_len -= 2;
                 } else if (phy_current_config.modulation == M_2FSK) {
-                    rx_data_length -= 4;
+                    cur_rx_packet_len -= 4;
                 }
-                device_driver.phy_rx_cb(rx_buffer, rx_data_length, 0xf0, rssi, rf_radio_driver_id);
+                device_driver.phy_rx_cb(rx_buffer, cur_rx_packet_len, 0xf0, rssi, rf_radio_driver_id);
                 // If auto ack used, must wait until RF returns to RF_TXPREP state
                 if ((version != MAC_FRAME_VERSION_2) && (rx_buffer[0] & FC_AR)) {
-                    wait_us(50);
+                    wait_us(100);
```

Comment:
is this time guess or data sheet defined

Reply:
Based on testing and measurements.

When data is received, RF state is very short time "TXPREP" before it starts auto-ack transmission and goes some other state. After ack transmission it goes back to "TXPREP".

So in here we wait 100us to give it time to start ack transmission and then we poll on the next line until it goes back to "TXPREP".

I had to make this wait longer because I optimised few SPI operations from the beginning of this function and it reached this wait_us much faster.